app/models/assessment.rb

```diff
@@ -1,14 +1,31 @@
 # coding: utf-8
 class Assessment < ApplicationRecord
 
-  belongs_to :format
-  belongs_to :created_by, class_name: 'Member'
+  belongs_to :member
+  has_many :questions, inverse_of: :assessment
+  has_many :other_versions,
+           -> (a) { where.not(id: a.id) },
+           class_name: 'Assessment',
+           foreign_key: :identifier, primary_key: :identifier
 
-  enum visibility: %i[internal external]
 
-  has_many :questions, inverse_of: :assessment
-  validates :format, :created_by, presence: true
+  enum visibility: %i[internal external]
 
   validates :version, uniqueness: { scope: :identifier }
 
+  before_validation :set_version, on: :create
+  validate :ensure_member_matches_other_versions
+
+  protected
+
+  def set_version
+    self.version = other_versions.size + 1
```

Comment:
Should we change `version` to be an integer?  It is still a string from when we let people set their own.